spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java

```diff
@@ -1202,34 +1198,20 @@ private Delivery consumeDelivery(Channel channel, String queueName, long timeout
 			}
 		}
 		catch (ExecutionException e) {
-			this.logger.error("Consumer failed to receive message: " + consumer, e.getCause());
-			exception = e.getCause();
+			Throwable cause = e.getCause();
+			this.logger.error("Consumer failed to receive message: " + consumer, cause);
+			exception = RabbitExceptionTranslator.convertRabbitAccessException(cause);
+			throw exception;
 		}
 		catch (InterruptedException e) {
 			Thread.currentThread().interrupt();
 		}
 		catch (TimeoutException e) {
 			// no result in time
```

Comment:
@artembilan @garyrussell also I'm not sure why the TimeoutException gets swallowed here and not for example rethrown as in `waitForConfirms`, same applies to InterruptedException, so I left it as is though there will be failing tests if this change will be performed. Just again the thought for consistency.

Reply:
I think this was decision of say first and made like this to treat these type of exceptions as "no result".

We are waiting for the `Delivery` object here and if we don't have it in time or our thread is interrupted, let's just return with `null`!

That's only an explanation and this comment confirm that.

We may reconsider this logic, but it is definitely a breaking for the current point version...

Reply:
@artembilan  thank you for elaborating on this one, more clear to me now.